src/ninja.cc

```diff
@@ -920,19 +955,50 @@ int NinjaMain::ToolCompilationDatabase(const Options* options, int argc,
 
   optind = 1;
   int opt;
-  while ((opt = getopt(argc, argv, const_cast<char*>("hx"))) != -1) {
+  const option long_options[] = {
+      {"target", required_argument, NULL, 't'},
+      {NULL, 0, NULL, 0}
+  };
+
+  std::vector<Edge*> edges_to_process = state_.edges_;
+  while ((opt = getopt_long(argc, argv, "x:", long_options, NULL)) != -1) {
     switch(opt) {
       case 'x':
         eval_mode = ECM_EXPAND_RSPFILE;
         break;
 
+      case 't': {
+        std::istringstream iss(optarg);
+        string target;
```

Comment:
nit: can we use std::string for all new code, since we care about removing all `using namespace std` in the future.